### DISCUSSION TITLE
my app crashes after kill from history when I try to rerun it on an iPhone (Android runs fine)

### DISCUSSION THREAD

**Discussion by mohammad-alshelleh** [firebase/flutterfire#4168] [Q&A]
I'm trying to build a simple app using Flutter for iOS and Android, but the app crashes on startup when I try to run it on an iPhone (Android runs fine).
I've been digging around some more and it looks like the crash is related to the Firebase dependencies. If I create the Flutter demo project (flutter create ), it runs fine on iOS. However if I then add Firebase packages to pubspec.yaml, and manually add GoogleServices-info.plist, the app crashes on rerun without android studio or xcode(from my iphone).

No error
flutter doctor:
[✓] Flutter (Channel master, 1.24.0-6.0.pre, on Mac OS X 10.15.7 19H2 darwin-x64, locale en-JO)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 12.1)
[✓] Android Studio (version 4.1)
[✓] Connected device (1 available)
! Error: iPhone is busy: Copying cache files from device. Xcode will continue when iPhone is finished. (code -10)

• No issues found!

**Comment by jamesblasco** [comment]
If your app works well when debugging and in release mode but it crashes when compiled in debug mode and launched without idle,  the problem is not related to Firebase.

The problem started on iOS 14 and will be fixed after iOS 14.2. 
iOS14 JIT problem: https://github.com/flutter/flutter/issues/61047
iOS14.2 JIT solution: https://github.com/flutter/flutter/issues/69984

**Reply by mohammad-alshelleh** [reply]
If I update the system to 14.2, will the problem be solved?
And if i upload the application on apple store , and someone downloads it in ios 14, the same problem will occur?

**Reply by jamesblasco** [reply]
No, they also have to do some changes in flutter. 

This problem only occurs in debug mode because flutter uses a different compiler(JIT: Just In Time). Building your app in release mode and publishing in the app store uses(AOT: Ahead of Time) and it will work perfectly fine.
